src/main/webapp/app/complaints/complaint-response.service.ts

```diff
@@ -51,39 +50,39 @@ export class ComplaintResponseService {
     createLock(complaintId: number): Observable<EntityResponseType> {
         return this.http
             .post<ComplaintResponse>(`${this.resourceUrl}/complaint/${complaintId}/create-lock`, {}, { observe: 'response' })
-            .pipe(map((res: EntityResponseType) => this.convertDateFromServer(res)));
+            .pipe(map((res: EntityResponseType) => this.convertComplaintResponseResponseDatesFromServer(res)));
     }
 
     resolveComplaint(complaintResponse: ComplaintResponse): Observable<EntityResponseType> {
-        const copy = this.convertDateFromClient(complaintResponse);
+        const copy = this.convertComplaintResponseDatesFromClient(complaintResponse);
         return this.http
             .put<ComplaintResponse>(`${this.resourceUrl}/complaint/${complaintResponse.complaint!.id}/resolve`, copy, { observe: 'response' })
-            .pipe(map((res: EntityResponseType) => this.convertDateFromServer(res)));
+            .pipe(map((res: EntityResponseType) => this.convertComplaintResponseResponseDatesFromServer(res)));
     }
 
     findByComplaintId(complaintId: number): Observable<EntityResponseType> {
         return this.http
             .get<ComplaintResponse>(`${this.resourceUrl}/complaint/${complaintId}`, { observe: 'response' })
-            .pipe(map((res: EntityResponseType) => this.convertDateFromServer(res)));
+            .pipe(map((res: EntityResponseType) => this.convertComplaintResponseResponseDatesFromServer(res)));
     }
 
-    public convertDateFromClient(complaintResponse: ComplaintResponse): ComplaintResponse {
+    public convertComplaintResponseDatesFromClient(complaintResponse: ComplaintResponse): ComplaintResponse {
         return Object.assign({}, complaintResponse, {
-            submittedTime: complaintResponse.submittedTime != undefined && dayjs(complaintResponse.submittedTime).isValid ? complaintResponse.submittedTime.toJSON() : undefined,
+            submittedTime: convertDateFromClient(complaintResponse.submittedTime),
         });
     }
 
-    public convertDateFromServer(res: EntityResponseType): EntityResponseType {
+    public convertComplaintResponseResponseDatesFromServer(res: EntityResponseType): EntityResponseType {
```

Comment:
the name of this methods sounds weird to be honest. Could you use a shorter one?

Reply:
I tried to use a constant naming scheme: "convert" + Objectname + "Response" if it's wrapped in an entity response + "DatesFromServer"
The only thing that could be left out is "Dates", everything else would leave out necessary Details.

I changed this occurrence to "convertComplaintResponseEntityResponseDatesFromServer",  which is sadly not shorter.

Can anyone else think of a better naming scheme?

Reply:
Can we not make one exception and make just this one shorter?
What about `convertComplaintResponseDatesFromServer`?

I think in general, we could leave out the "Response" and simply use "convert" + Objectname + "DatesFromServer"

Reply:
Many services have a method for the response and another method for the actual object. So `convertComplaintResponseDatesFromServer` already exists and overloading the method is not possible here

Reply:
then leave it like it is